File: leet-code/1. Two Sum/TwoSum1_2.java
import java.util.Arrays;
import java.util.HashMap;

public class TwoSum1_2 {
    public int[] twoSum(int[] nums, int target) {
        // 숫자, 인덱스
        HashMap<Integer, Integer> maps = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            maps.put(nums[i], i);
        }
        int first = -1;
        int second = -1;
        for (int i = 0; i < nums.length; i++) {
            // 키 확인 && 값 확인
            if (maps.containsKey(target - nums[i]) && i != maps.get(target - nums[i])) {
                first = i;
                second = maps.get(target - nums[i]);
                break;
            }
        }
        return new int[]{first, second};
    }
}

File: leet-code/1. Two Sum/TwoSum1_2.kt
package com.example.algorithm

class TwoSum1_2 {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val sortedNums = nums
            .mapIndexed { index, value -> intArrayOf(value, index)}
            .sortedBy { it[0] }
        var left = 0
        var right = nums.size - 1
        while (left < right) {
            val sums = sortedNums[left][0] + sortedNums[right][0]
            if (sums == target) {
                break
            }
            if (sums < target) {
                left += 1
            } else {
                right -= 1
            }
        }
        return intArrayOf(sortedNums[left][1], sortedNums[right][1])
    }
}
